src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -398,6 +401,7 @@ private async Task UnsynchronizedStartHostAsync(ScriptHostStartupOperation activ
                     logger.ErrorOccurredInactive(activeOperation.Id, exc);
                 }
 
+                _hostMetrics.AppFailure();
```

Comment:
When in a crash loop, attemptCount will keep increasing, but will be reset back to zero after the first successful startup. Won't your new metric just monotonically increase? Is the idea that it's up to the consumer of the metric to do diffs at different points in time to determine when the app is healthy? How long to wait between samples - e.g. we have an exponential backoff on retries, so they'll eventually slow down to once every 2 minutes.

Reply:
Hmm that's a good point. I think this would be okay? The consumer would be getting a time series of metrics every 30 seconds or so, and the metrics provider clears the cache every time we write to the metric file via the flex publisher. The consumer would be able to see if this goes back to 0 for a considerate amount of time. How often do we retry and do we ever stop retrying?

On the consumer side (this is the main use case), if this value is > 1 the instance/pod is signaled as unhealthy and there is action to recycle. If every instance in a site has this value > 1 - then we scale down to 1 instance until we see this value at 0 for a couple minutes. 

I'm open to suggestions here on how to best set this up to enable the use case I mention